agent/bench-scripts/postprocess/uperf-postprocess

```diff
@@ -192,55 +204,99 @@ foreach $result_file (sort readdir($dh) ) {
 						# the latency is the full round trip for 1 transaction, which is also = ($instances / trans_sec)
 						# if --log-response-times in pbench_uperf is used, the write and read latency can be computed (later).
 						if ( $nr_ops_diff != 0) {
-							my %latency_sample = ( get_label('date_label') => int $timestamp_ms, get_label('value_label') => $instances /$trans_sec *1000000);
-							push(@latency_samples, \%latency_sample);
+							$latency_samples{$timestamp_ms} = {
+								get_label('date_label')  => int $timestamp_ms,
+								get_label('value_label') =>
+									$instances / $trans_sec * 1000000
+							};
 						}
-						my %throughput_sample = (get_label('date_label') => int $timestamp_ms, get_label('value_label') => $trans_sec);
-						push(@throughput_samples, \%throughput_sample);
+						$throughput_samples{$timestamp_ms} = {
+							get_label('date_label')  => int $timestamp_ms,
+							get_label('value_label') => $trans_sec
+						};
 					} else {
-						my $bytes_sec = $nr_bytes_diff /$timestamp_s_diff;
-						my %throughput_sample = (get_label('date_label') => int $timestamp_ms, get_label('value_label') => ($nr_bytes_diff *8 /1000000000 /$timestamp_s_diff));
-						push(@throughput_samples, \%throughput_sample);
+						$throughput_samples{$timestamp_ms} = {
+							get_label('date_label')  => int $timestamp_ms,
+							get_label('value_label') =>
+								($nr_bytes_diff * 8 / 1000000000 / $timestamp_s_diff)
+						};
 					}
 				}
-				$prev_timestamp_ms = $timestamp_ms;
+				$prev{'timestamp'} = $timestamp_ms;
+				$prev{'ops'} = $ops;
+				$prev{'bytes'} = $bytes;
 			}
 		}
 		$nr_result_files++;
 		close(IN_FILE);
-		# the uid should be unique
-		if ( (@latency_samples) ) {
-			trim_series(\@latency_samples, $series_trim, $series_trim);
+
+		# Trim the last several seconds of data from the series.
+		my $limit = $prev{'timestamp'} - $series_trim * 1000;
+		foreach my $ts (sort {$b <=> $a} keys %latency_samples) {  # Reverse-order
+			if ($ts <= $limit) {
+				last;
+			}
+			delete $latency_samples{$ts};
+		}
+
+		if (%latency_samples) {
 			my %this_lat_dataset;
 			my $description = "Average elapsed time spanning: client sending, server accepting/sending, and client receiving 1 message, over a 1 second window";
-			my $mean = get_mean(\@latency_samples);
-			%this_lat_dataset = ( 	get_label('uid_label') => create_uid('client_hostname_label', 'server_hostname_label', 'server_port_label'), get_label('description_label') => $description,
-						get_label('role_label') => $role, get_label('client_hostname_label') => $client_hostname,
-						get_label('server_hostname_label') => $server_hostname, get_label('server_port_label') => "$server_port",
-						get_label('value_label') => $mean, get_label('timeseries_label') => \@latency_samples   );
+			my $mean = get_mean_hash(\%latency_samples);
+			%this_lat_dataset = (
```

Comment:
Did you want to do here what you did in linpack-postprocess and push the value directly without storing it first?

Reply:
Done.